### ISSUE TITLE
Error while compiling ( v run blog.v ) 

### ISSUE THREAD

**Issue by B3ns44d** [vlang/v#7838]
**V version:**  v_linux.zip 0.2.1
**OS:** Kali Linux

**What did you do?**

I'm trying to run blog.v from the tutorial using `v run blog.v`

**What did you expect to see?**

**Hello, world from vweb!** on the browser 
 
**What did you see instead?**

> /home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:206:6: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  204 |     println('Running a Vweb app on http://localhost:$port')
  205 |     l := net.listen_tcp(port) or { panic('failed to listen') }
  206 |     app.vweb = Context{}
      |         ~~~~
  207 |     app.init_once()
  208 |     $for method in T.methods {
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:318:6: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  316 |     }
  317 |     // mut app := T{
  318 |     app.vweb = Context{
      |         ~~~~
  319 |         req: req
  320 |         conn: conn
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:322:21: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  320 |         conn: conn
  321 |         form: map[string]string{}
  322 |         static_files: app.vweb.static_files
      |                           ~~~~
  323 |         static_mime_types: app.vweb.static_mime_types
  324 |         page_gen_start: page_gen_start
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:323:26: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  321 |         form: map[string]string{}
  322 |         static_files: app.vweb.static_files
  323 |         static_mime_types: app.vweb.static_mime_types
      |                                ~~~~
  324 |         page_gen_start: page_gen_start
  325 |     }
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:328:7: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  326 |     // }
  327 |     if req.method in methods_with_form {
  328 |         app.vweb.parse_form(req.data)
      |             ~~~~
  329 |     }
  330 |     if vals.len < 2 {
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:338:30: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  336 |     // Serve a static file if it is one
  337 |     // TODO: handle url parameters properly - for now, ignore them
  338 |     mut static_file_name := app.vweb.req.url
      |                                 ~~~~
  339 |     if static_file_name.contains('?') {
  340 |         static_file_name = static_file_name.all_before('?')
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:340:20: error: assignment mismatch: 1 variable(s) but `all_before()` returns 0 value(s)
  338 |     mut static_file_name := app.vweb.req.url
  339 |     if static_file_name.contains('?') {
  340 |         static_file_name = static_file_name.all_before('?')
      |                          ^
  341 |     }
  342 |     static_file := app.vweb.static_files[static_file_name]
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:342:21: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  340 |         static_file_name = static_file_name.all_before('?')
  341 |     }
  342 |     static_file := app.vweb.static_files[static_file_name]
      |                        ~~~~
  343 |     mime_type := app.vweb.static_mime_types[static_file_name]
  344 |     if static_file != '' && mime_type != '' {
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:343:19: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  341 |     }
  342 |     static_file := app.vweb.static_files[static_file_name]
  343 |     mime_type := app.vweb.static_mime_types[static_file_name]
      |                      ~~~~
  344 |     if static_file != '' && mime_type != '' {
  345 |         data := os.read_file(static_file) or {
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:344:5: error: left operand for `&&` is not a boolean
  342 |     static_file := app.vweb.static_files[static_file_name]
  343 |     mime_type := app.vweb.static_mime_types[static_file_name]
  344 |     if static_file != '' && mime_type != '' {
      |        ~~~~~~~~~~~~~~~~~
  345 |         data := os.read_file(static_file) or {
  346 |             send_string(conn, http_404) or { }
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:349:7: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  347 |             return
  348 |         }
  349 |         app.vweb.send_response_to_client(mime_type, data)
      |             ~~~~
  350 |         data.free()
  351 |         return
/home/MYUSERNAME/Desktop/vlang/v/vlib/vweb/vweb.v:375:10: error: type `App` has no field or method `vweb`.
1 possibility: `Context`.
  373 |             for data in tmp_query {
  374 |                 if data.len == 2 {
  375 |                     app.vweb.query[data[0]] = data[1]
      |                         ~~~~
  376 |                 }
  377 |             }

When I run ` v install vweb` I see this

> cannot compile `/home/MYUSERNAME/Desktop/vlang/v/cmd/tools/vpm.v`: 
V panic: failed to open file "/tmp/v/vpm.2915694233743800336.tmp.c"
                                                        | 0x55c5d7cbabde | /home/MYUSERNAME/Desktop/vlang/v/v(+0xcbde) 
                                                        | 0x7f1405140cca | /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xea) 
                                                        | 0x55c5d7cbac9a | /home/MYUSERNAME/Desktop/vlang/v/v(+0xcc9a)

**Comment by JalonSolov** [comment]
Try running `v up` then try the blog example again.

**Comment by B3ns44d** [comment]
this what happens when I run `v up `

> youssef@b3ns44d:~/Desktop/vlang/v/tutorials/code/blog$ v up
cannot compile `/home/youssef/Desktop/vlang/v/cmd/tools/vup.v`: 
V panic: failed to open file "/tmp/v/vup.16795370892768379802.tmp.c"
                                                        | 0x55c6c5930bde | /home/youssef/Desktop/vlang/v/v(+0xcbde) 
                                                        | 0x7f92ec031cca | /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xea) 
                                                        | 0x55c6c5930c9a | /home/youssef/Desktop/vlang/v/v(+0xcc9a)

**Comment by JalonSolov** [comment]
Ok, that's odd, but...  try this instead:
```
cd /home/youssef/Desktop/vlang && git clone https://github.com/vlang/v && cd v && make && cd ~/Desktop/vlang/v/tutorials/code/blog
```
then try again.

**Comment by B3ns44d** [comment]
I've already tried this before and have the same problem

> cannot compile `/home/youssef/Desktop/vlang/v/cmd/tools/vself.v`: 
V panic: failed to open file "/tmp/v/vself.9129665764552531223.tmp.c"
                                                        | 0x5641bb4c417d | /home/youssef/Desktop/vlang/v/v(+0x14717d) 
                                                        | 0x5641bb4c468e | /home/youssef/Desktop/vlang/v/v(+0x14768e) 
                                                        | 0x5641bb4ce530 | /home/youssef/Desktop/vlang/v/v(+0x151530) 
                                                        | 0x5641bb4d6df1 | /home/youssef/Desktop/vlang/v/v(+0x159df1) 
                                                        | 0x5641bb4e2dbc | /home/youssef/Desktop/vlang/v/v(+0x165dbc) 
                                                        | 0x7f9e6f129cca | /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xea) 
                                                        | 0x5641bb37f5ba | /home/youssef/Desktop/vlang/v/v(+0x25ba) 

make: *** [Makefile:89: all] Error 1

**Comment by fnetX** [comment]
@B3ns44d To the actual problem: vweb has been updated to use an embedded struct, the examples in this repo are already using the next vweb version, you'll only be able to run them with the latest compiler. 
I think you could alternatively checkout e.g. a9ab79d30 and use these examples and this vweb version.

**Comment by B3ns44d** [comment]
@fnetX I don't the problem from `vweb` as you can see the error message says that cannot compile `vself.v ` file because  of `V panic: failed to open file "/tmp/v/vself.9129665764552531223.tmp.c"` I am using the latest version available now which is the `v_linux.zip 0.2.1` I think the problem is in these two files the `vself.v` file or `vself.9129665764552531223.tmp.c`

**Comment by fnetX** [comment]
Your original issue:
```
error: type App has no field or method vweb.
1 possibility: Context.
```
comes from the change in the default struct. It was: 
```
App {
vweb vweb.Context
}
```
and is now:
```
App {
vweb.Context
}
```
So the example does no longer work. But you're right, your issue to install vweb is another one, indeed.

**Comment by fnetX** [comment]
So if you just want to try V and run the blog example, you can install the latest stable V that works for you and use the tutorial from the same repo state (= an older commit), so you can play around with vweb and the blog until the second issue is fixed.

**Comment by B3ns44d** [comment]
Same issues with blog example and another issue with the compiler I'm trying to play with V Weekly. 2020.51.2 and I've gotten these issues instead

```
cannot compile `/home/youssef/Desktop/vlang/v/cmd/tools/vself.v`: 
vlib/os/args.v:14:9: error: use `array2 = array1.clone()` instead of `array2 = array1` (or use `unsafe`)
   12 |     mut cargs := []string{}
   13 |     if cut_word !in args {
   14 |         cargs = args
      |               ^
   15 |     } else {
   16 |         mut found := false
vlib/os/args.v:40:9: error: use `array2 = array1.clone()` instead of `array2 = array1` (or use `unsafe`)
   38 |     mut cargs := []string{}
   39 |     if cut_word !in args {
   40 |         cargs = args
      |               ^
   41 |     } else {
   42 |         cargs << args[0]

make: *** [Makefile:63: all] Error 1

```

First I got this error message and fixed it by changing `14 |         cargs = args` **to ==>** `14 |         cargs = args.clone()` and now I have a new issue

```
youssef@b3ns44d:~/Desktop/vlang/v$ make
cd ./vc && git clean -xf && git pull --quiet
cd ./thirdparty/tcc && git clean -xf && git pull --quiet
cc  -g -std=gnu99 -w -o ./v ./vc/v.c -lm -lpthread 
./v self
cannot compile `/home/youssef/Desktop/vlang/v/cmd/tools/vself.v`: 
==================
      |                                 new_map_1
/tmp/v/vself.15073642771474637749.tmp.c:2775:73: error: ‘map_hash_string’ undeclared (first use in this function); did you mean ‘map_string_string’?
 2775 |  map_string_int first_bigrams = new_map_2(sizeof(string), sizeof(int), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string);
      |                                                                         ^~~~~~~~~~~~~~~
      |                                                                         map_string_string
/tmp/v/vself.15073642771474637749.tmp.c:2775:73: note: each undeclared identifier is reported only once for each function it appears in
/tmp/v/vself.15073642771474637749.tmp.c:2775:91: error: ‘map_eq_string’ undeclared (first use in this function)
 2775 |  map_string_int first_bigrams = new_map_2(sizeof(string), sizeof(int), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string);
      |                                                                                           ^~~~~~~~~~~~~
/tmp/v/vself.15073642771474637749.tmp.c:2775:107: error: ‘map_clone_string’ undeclared (first use in this function); did you mean ‘map_string_string’?
 2775 |  map_string_int first_bigrams = new_map_2(sizeof(string), sizeof(int), &map_hash_string, &map_eq_string, &map_clone_string, &map_free_string);
      |                                                                                                           ^~~~~~~~~~~~~~~~
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

If you were not working with C interop, please raise an issue on GitHub:

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang


make: *** [Makefile:63: all] Error 1


```

**Comment by medvednikov** [comment]
try using V from source, it's actually easier than using prebuilt packages

git clone https://github.com/vlang/v && cd v && make

**Comment by B3ns44d** [comment]
**Well now everything works**
